src/pages/index.js

```diff
@@ -68,51 +98,138 @@ cardPreviewPopup.close();
 /*                                Card Section                                */
 /* -------------------------------------------------------------------------- */
 
+
 // This function creates a new card
-function createCard(data) {
-    const cardElement = new Card({ data, handleImageClick: (imageData) => {
-        cardPreviewPopup.open(imageData);
+function createCard(data, userId) {
+    const cardElement = new Card({ 
+        data,
+        handleImageClick: (imageData) => {
 
-    }}, selectors.cardTemplate);
+            // Open image popup on click
+            cardPreviewPopup.open(imageData);
+        },
+        handleDeleteClick: () => {
+            
+            // Create the delete popup instance
+            const deletePopup = new PopupWithConfirm(selectors.deletePopup, () => {
 
-    return cardElement.getView();
-}
+                // Handle card deletion
```

Comment:
You should also notify the user about loading by changing button text

Reply:
Done!